pkg/image/apis/image/types.go

```diff
@@ -469,6 +469,36 @@ type ImageStreamImage struct {
 // DockerImageReference points to a Docker image.
 type DockerImageReference = reference.DockerImageReference
 
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ImageStreamLayers describes information about the layers referenced by images in this
+// image stream.
+type ImageStreamLayers struct {
+	metav1.TypeMeta
+	// Standard object's metadata.
+	metav1.ObjectMeta
+	// blobs is a map of blob name to metadata about the blob.
+	Blobs map[string]ImageLayerData
+	// layers is a map between an image name and the names of the blobs that form the
+	// images.
+	Layers map[string]BlobReferences
+	// manifests is a map between an image name and the name of the manifest blob, if one
+	// exists.
+	Manifests map[string]string
+}
+
+// BlobReferences is an array of names of blobs.
+type BlobReferences []string
+
+// ImageLayerData contains metadata about an image layer.
+type ImageLayerData struct {
+	// Size of the layer in bytes as defined by the underlying store. This field is
+	// optional if the necessary information about size is not available.
+	LayerSize *int64
```

Comment:
Do we have an immediate use-case for this?

Reply:
On registry side this information will be cached once again. Maybe it's better to make a watcher on registry side and cache it just once per registry ?

Reply:
I figured, I was just expressing a reason it might be worth paying that cost.  (Ultimately I think @smarterclayton convinced himself that this cache won't be that big)